core/amber/src/test/scala/edu/uci/ics/texera/web/MockTexeraDB.scala

```diff
@@ -0,0 +1,66 @@
+package edu.uci.ics.texera.web
+
+import ch.vorburger.mariadb4j.DB
+import com.mysql.cj.jdbc.MysqlDataSource
+import org.jooq.DSLContext
+import org.jooq.impl.DSL
+import ch.vorburger.mariadb4j.{DB, DBConfigurationBuilder}
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Path, Paths}
+
+trait MockTexeraDB {
+
+  private var dbInstance: Option[DB] = None
+  private var dslContext: Option[DSLContext] = None
+
+  def getDSLContext: DSLContext = {
+    dslContext match {
+      case Some(value) => value
+      case None =>
+        throw new RuntimeException(
+          "test database is not initialized. Did you call initializeDBAndReplaceDSLContext()?"
+        )
+    }
+  }
+
+  def shutdownDB(): Unit = {
+    dbInstance match {
+      case Some(value) =>
+        value.stop()
+        dbInstance = None
+        dslContext = None
+      case None =>
+      // do nothing
+    }
+  }
+
+  def initializeDBAndReplaceDSLContext(): Unit = {
+    assert(dbInstance.isEmpty && dslContext.isEmpty)
+    val database: String = "texera_db"
+    val username: String = "root"
+    val password: String = ""
+    val ddlPath =
+      Paths.get("..").resolve("scripts").resolve("sql").resolve("texera_ddl.sql").toRealPath()
+    val content = new String(Files.readAllBytes(ddlPath), StandardCharsets.UTF_8)
+    val config = DBConfigurationBuilder.newBuilder
+      .setPort(0) // 0 => automatically detect free port
+      .addArg("--default-time-zone=+0:00")
```

Comment:
In ddl and production DB instances, I think we are using PST (GMT -8)? Maybe we should mock the same?

Reply:
done